portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,20 +29,35 @@
 
 /** Servlet that allows users to add comments and returns comment history  */
 @WebServlet("/comments")
-public class DataServlet extends HttpServlet {
-    
+public class DataServlet extends HttpServlet {  
+  
+  private static int maxComments;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    String maxCommentsString = request.getParameter("numcomments");
+
+    // Convert the input to an int.
+    try {
+        maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+        System.err.println("Could not convert to int: " +  maxCommentsString);
```

Comment:
What happens to maxComments after we print this error and continue on with the method?

Reply:
I initially added this when I had initialized maxComments to 10  as the default number of comments but I changed this when I added an argument to the getComments(numComments) function in javascript and instead used the `<body onload="getComments(10)">`line in index.html to set the default number of comments.
When maxComments was initialized to 10, maxComments would keep this value if there was a NumberFormatException.
Now that you ask this I'm wondering whether I need to include this try-catch statement at all if the String maxCommentsString comes directly from the options in the dropdown menu, so the catch might never execute.
If it's better to keep the try-catch statement regardless I could have the catch statement assign 0 to maxComments.

Are there any best practices design-wise for this situation?

Reply:
Best practice: never trust data from the user :) 

The select box has limited options, but it's easy to manipulate them with, e.g., the inspector, or to intercept the `fetch` requests that the browser makes (on behalf of the user, after all) and modify them. So you always want to cleanse user input the best you can, such as doing try-catch here.

So, I wasn't able to create an exception in my testing, but yes, I would set a default value in the catch block if `parseInt` fails. Probably set it to your default of 10, though, rather than zero, so that the handler fails gracefully and still does something useful.

Reply:
Sounds good! I'll set it to the default value - I might change this to 5 now that the comment boxes take up more space